main/src/cgeo/geocaching/settings/TemplateTextPreference.java

```diff
@@ -80,9 +90,10 @@ private void insertSignatureTemplate(final LogTemplate template) {
     @Override
     protected void onDialogClosed(final boolean positiveResult) {
         if (positiveResult) {
-            final String text = editText.getText().toString();
-            persistString(text);
-            callChangeListener(text);
+            Settings.putLogTemplate(preferenceBaseName, title.getText().toString(), editText.getText().toString());
+            String displayValue = title.getText().toString() + ": " + editText.getText().toString();
+            persistString(displayValue);
```

Comment:
Wh is this still needed?

Reply:
refer to the answer about the third property - it's what's displayed in the prefs dialog. Those lines are needed for that dialog to update properly, without them it only updates when calling the Activity again.